From 0f90d03ae1fb2a11b2d5de958abbc1fd57b9af5e Mon Sep 17 00:00:00 2001
From: "Zhong,Fangjian" <fangjian.zhong@intel.com>
Date: Wed, 21 Jun 2023 01:58:20 +0800
Subject: [PATCH 1/2] Update bootcontrol from HIDL to AIDL

android.hardware.boot@1.2::IBootControl/default is deprecated in
compatibility matrix at FCM Version 8, update bootcontrol to AIDL
implementation.

Tracked-On: OAM-110593
Signed-off-by: Zhong,Fangjian <fangjian.zhong@intel.com>
---
 groups/device-specific/caas/framework_manifest.xml | 9 ---------
 groups/device-specific/caas/manifest.xml           | 9 ---------
 groups/slot-ab/true/product.mk                     | 5 ++---
 3 files changed, 2 insertions(+), 21 deletions(-)

diff --git a/groups/device-specific/caas/framework_manifest.xml b/groups/device-specific/caas/framework_manifest.xml
index 3ee7577..312168f 100644
--- a/groups/device-specific/caas/framework_manifest.xml
+++ b/groups/device-specific/caas/framework_manifest.xml
@@ -182,15 +182,6 @@
             <instance>external/0</instance>
         </interface>
     </hal>
-    <!--hal format="hidl">
-        <name>android.hardware.boot</name>
-        <transport>hwbinder</transport>
-        <version>1.2</version>
-        <interface>
-            <name>IBootControl</name>
-            <instance>default</instance>
-        </interface>
-    </hal-->
     <hal format="hidl">
         <name>android.hardware.renderscript</name>
         <transport arch="32+64">passthrough</transport>
diff --git a/groups/device-specific/caas/manifest.xml b/groups/device-specific/caas/manifest.xml
index 54bb15b..019f754 100644
--- a/groups/device-specific/caas/manifest.xml
+++ b/groups/device-specific/caas/manifest.xml
@@ -149,15 +149,6 @@
             <instance>legacy/0</instance>
         </interface>
     </hal>
-    <!--hal format="hidl">
-        <name>android.hardware.boot</name>
-        <transport>hwbinder</transport>
-        <version>1.2</version>
-        <interface>
-            <name>IBootControl</name>
-            <instance>default</instance>
-        </interface>
-    </hal-->
     <hal format="hidl">
         <name>android.hardware.renderscript</name>
         <transport arch="32+64">passthrough</transport>
diff --git a/groups/slot-ab/true/product.mk b/groups/slot-ab/true/product.mk
index 21e2f15..6e85977 100644
--- a/groups/slot-ab/true/product.mk
+++ b/groups/slot-ab/true/product.mk
@@ -5,9 +5,8 @@ PRODUCT_PACKAGES += \
     libavb \
     update_engine_sideload \
     avbctl \
-    android.hardware.boot@1.2-impl-intel \
-    android.hardware.boot@1.2-impl-intel.recovery \
-    android.hardware.boot@1.2-service \
+    android.hardware.boot-service.intel \
+    android.hardware.boot-service.recovery \
     bootctrl.intel \
     bootctrl.intel.recovery
 
-- 
2.25.1

